src/Entity/AlertBannerEntity.php

```diff
@@ -406,7 +409,21 @@ public function getCacheContexts() {
       $this->addCacheContexts($contexts);
     }
 
-    return parent::getCacheContexts();
+    return Cache::mergeContexts(parent::getCacheContexts(), $contexts);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+
+    $cache_tags = [];
+
+    // Get token and use as a cache tag.
+    $token = $this->getToken();
+    $cache_tags = Cache::mergeTags($cache_tags, ['localgov.alert.banner.token:' . $token]);
+
+    return Cache::mergeTags(parent::getCacheTags(), $cache_tags);
```

Comment:
If this is called multiple times then a new `$token` will be generated each time. Might be better to move this to the constructor and add the cache tag using the `addCacheTags` method.

Reply:
I thought the token doesn't change unless you update an alert banner.

Reply:
Yes the token won't change unless the banner is updated. 
Can we call methods in the constructor?